packages/@react-spectrum/s2/src/ActionButton.tsx

```diff
@@ -79,9 +80,21 @@ export const btnStyles = style<ButtonRenderProps & ActionButtonStyleProps & Togg
     isJustified: 0
   },
   fontWeight: 'medium',
+  width: 'fit',
   userSelect: 'none',
   transition: 'default',
   forcedColorAdjust: 'none',
+  position: 'relative',
+  gridTemplateAreas: {
+    default: ['icon text'],
+    [iconOnly]: ['icon'],
+    [textOnly]: ['text']
+  },
+  gridTemplateColumns: {
+    default: ['auto', 'auto'],
```

Comment:
```suggestion
    default: [minmax('fit', 'auto), minmax('fit', 'auto)],
```
I think you might need to do something like this, try the ActionButton "resizing" story

Reply:
it might fix the ToggleButton chromatic issue as well